src/Models/AuditActivity.php

```diff
@@ -43,20 +46,20 @@ class AuditActivity extends BaseModel
         'ip_address'
     ];
 
-    protected $dates = [
-        'created_at'
+    protected $casts = [
+        'created_at' => 'datetime'
     ];
 
     //--------------------------------------------------------------------------------------------------------------
     // Mutators
     //--------------------------------------------------------------------------------------------------------------
 
-    protected function setIpAddressAttribute($value)
+    protected function setIpAddressAttribute(string|null $value): void
```

Comment:
Nullable type hint here too?
```suggestion
    protected function setIpAddressAttribute(?string $value): void
```

Reply:
Is this just sugar? Or is anything different from the two? Just curious as `?|string` is not a valid docblock, but `string|null` is.

Reply:
It's syntactic sugar
> A single base type declaration can be marked nullable by prefixing the type with a question mark (?). Thus ?T and T|null are identical.

Reply:
Okay, so it seems it works in userland, just docblock parsing doesn't work yet for those - https://github.com/laravel/framework/pull/45677/files